src/ProjectTemplates/Web.ProjectTemplates/content/BlazorWeb-CSharp/BlazorWeb-CSharp/Identity/IdentityRevalidatingAuthenticationStateProvider.cs

```diff
@@ -0,0 +1,54 @@
+using System.Security.Claims;
+using Microsoft.AspNetCore.Components.Authorization;
+using Microsoft.AspNetCore.Components.Server;
+using Microsoft.AspNetCore.Identity;
+using Microsoft.Extensions.Options;
+using BlazorWeb_CSharp.Data;
+
+namespace BlazorWeb_CSharp.Identity;
+
+public class IdentityRevalidatingAuthenticationStateProvider : RevalidatingServerAuthenticationStateProvider
```

Comment:
Would be good to add a comment block above this class to indicate the basic ideas about what it does.